src/components/buttons/button.tsx

```diff
@@ -30,17 +44,21 @@ export const Button = ({ children, icon, disable, compact, type, paddingHorizont
     ...buttonStyleMapping[type],
     height: compact ? '40px' : baseButtonStyles.height,
     padding: paddingHorizontal?.toString() || baseButtonStyles.padding,
-    minWidth: expand ? '100%' : baseButtonStyles.minWidth,
+    width: expand ? '100%' : 'auto',
   };
 
   const buttonHover = type === 'primary' && !disable ? 'primary-button' : '';
 
   return (
     <div>
       <button className={buttonHover} style={buttonStyle} disabled={disable} onClick={onClick}>
-        {!!icon && <span>{icon}</span>}
-        <Separator horizontal size={8} />
-        <label> {children} </label>
+        {!!icon && (
+          <>
+            <span>{icon}</span>
+            {icon && title && <Separator horizontal size={8} />}
+          </>
+        )}
+        <LargeLabel weight={titleWeight} color={color} title={title} />
```

Comment:
humm, acho que o Label pode manter recebendo a string como child, mas o botao nao precisaria, até porque a Label poderia ser usada em outros lugares sem depender do botao, ne?

```suggestion
        <LargeLabel weight={titleWeight} color={color}>{title}<LargeLabel/>
```